### PR TITLE
Move security guidelines

### PR DESCRIPTION
**Pull request by GotenJBZ**
### Description
add Move Security Guidelines.
### Checklist

- [x] Do all Lints pass?
- [x] Have you ran `pnpm spellcheck`?
- [x] Have you ran `pnpm fmt`?
- [x] Have you ran `pnpm lint`?

### CODE REVIEW

File: docs/move/move-on-aptos/move-security-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,574 @@
+---
+title: "Move Security Guidelines"
```
Comment (gedigi) [inline docs/move/move-on-aptos/move-security-guidelines.md:2]: I would just call it "Security" to align with the other sections

File: docs/move/move-on-aptos/move-security-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+public fun repay_flash_loan<T>(rec: Receipt, coins: Coin<T>) {
+    let Receipt{ amount } = rec;
+    assert!(coin::value<T>(&coin) >= rec.amount, 0);
+    deposit(coin);
+}
+```
+
+### Example Secure Code
+
+The Aptos Framework sample below creates a key-value table consisting of two generic types `K` and `V` . Its related `add` functions accepts as parameters a `Table<K, V>` object, a `key`, and a `value` of types `K` and `V` . The `phantom` syntax ensures that the key and value types cannot be different than those in the table, preventing type mismatches. [Read more](https://diem.github.io/move/generics.html#phantom-type-parameters) about `phantom` type parameters.
```
Comment (gedigi) [inline docs/move/move-on-aptos/move-security-guidelines.md:159]:
```suggestion
The Aptos Framework sample below creates a key-value table consisting of two generic types `K` and `V` . Its related `add` functions accepts as parameters a `Table<K, V>` object, a `key`, and a `value` of types `K` and `V` . The `phantom` syntax ensures that the key and value types cannot be different than those in the table, preventing type mismatches. [Read more](https://aptos.dev/move/book/generics/#phantom-type-parameters) about `phantom` type parameters.
```

File: docs/move/move-on-aptos/move-security-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+
+# Move Security Guidelines
+
+# Introduction
+
+The Move language is designed with security and inherently offers several features including a type system and a linear logic. Despite this, its novelty and the intricacies of some business logic mean that developers might not always be familiar with Move's secure coding patterns, potentially leading to bugs.
+
+This guide addresses this gap by detailing common anti-patterns and their secure alternatives. It provides practical examples to illustrate how security issues can arise and recommends best practices for secure coding. This guide aims to sharpen developers' understanding of Move's security mechanisms and ensure the robust development of smart contracts.
+
+# Access Control
```
Comment (gedigi) [inline docs/move/move-on-aptos/move-security-guidelines.md:13]: considering the right column indexes from H2 onward, would it make sense to shift all the headers by 1? so `Access Control` is `##`, `Object Access Control` is `###` and so forth

**Review by gedigi — APPROVED** [review]

File: docs/move/move-on-aptos/move-security-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+| private        | ✅            | ⛔                                | ⛔            |
+| public(friend) | ✅            | ✅ if friend<br></br>⛔ otherwise | ⛔            |
+| public         | ✅            | ✅                                | ⛔            |
+| entry          | ✅            | ⛔                                | ✅            |
+
+This layered visibility ensures that only authorized entities can execute certain functions, greatly reducing the risk of bugs or attacks that compromise contract integrity.
+
+Note that it’s possible to combine `entry` with `public` or `public(friend)`
+
+```rust
```
Comment (gregnazario) [inline docs/move/move-on-aptos/move-security-guidelines.md:108]: You can change all of these to as it's supported in github
````
```move
...
```
````

**Pull request merged**